tap_gitlab/client.py

```diff
@@ -24,11 +27,21 @@ class GitLabStream(RESTStream):
     next_page_token_jsonpath = "$.X-Next-Page"
     extra_url_params: dict = {}
     bookmark_param_name = "since"
+    _LOG_REQUEST_METRIC_URLS = True  # Okay to print in logs
+    # sensitive_request_path = False  # TODO: Update SDK to accept this instead.
 
     @property
     def url_base(self) -> str:
-        """Return the API URL root, configurable via tap settings."""
-        return self.config.get("api_url", DEFAULT_API_URL)
+        """Return the API URL root, configurable via tap settings.
+
+        If no path is provided, the base URL will be appended with `/api/v4`.
+        E.g. 'https://gitlab.com' would become 'https://gitlab.com/api/v4'
+        """
+        result = self.config.get("api_url", DEFAULT_API_URL)
+        if "/" not in result.replace("://", ""):
```

Comment:
Wouldn't this break if I provided `https://gitlab.com/` as a config value?

Reply:
@laurentS - Yes, that would break. I am open to alternative approaches or heuristics.

I originally expected we'd requite the exact url base (or leave blank to use the default) but our legacy `discovery.yml` had passed just `https://meltano.com` so we'll have to handle that case at least.

https://gitlab.com/meltano/meltano/blob/master/src%2Fmeltano%2Fcore%2Fbundle%2Fdiscovery.yml#L975-977

Reply:
```suggestion
        if "/" not in result.replace("://", "").rstrip('/ '):
```
How about something like this? That's assuming the api url is never the hostname alone, but I guess that's a reasonable assumption.

Reply:
Because this tap expects `path` to always have a preceding `/` (for instance, `/projects`), we can `rstrip('/')` from base URL if we want to be tolerant of both `https://gitlab.com` and also `https://gitlab.com/`.

Reply:
https://github.com/MeltanoLabs/tap-gitlab/pull/77/commits/a426d4ff3d9798aacdc06d54470a0083e6acca40

Reply:
Seems ok. As an alternative, using `urllib` to provide a bit more resilience (vs working with strings) could work as well:
```python
>>> from urllib.parse import urlparse
>>> o = urlparse("https://gitlab.com/")
>>> o.path
'/'
>>> o = urlparse("https://gitlab.com")
>>> o.path
''
>>> o = urlparse("https://gitlab.com/api/v4")
>>> o.path
'/api/v4'
>>> o = urlparse("https://gitlab.com/api/v4/")
>>> o.path
'/api/v4/'
```
so maybe something like `if len(urlparse(result)) <= 1:` ?
Though I agree that normalising/validating config params as early as possible seems like a good option to avoid questions like this.

Reply:
Updated to use urlparse(...).path in this check. Thanks, @laurentS